runtime/src/main/java/com/dylibso/chicory/runtime/Memory.java

```diff
@@ -1,363 +1,106 @@
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public void writeByte(int addr, byte data) {
-        try {
-            buffer.put(addr, data);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public long readU8(int addr) {
-        try {
-            return read(addr) & 0xFF;
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public long readI8(int addr) {
-        try {
-            return read(addr);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public void writeF32(int addr, float data) {
-        try {
-            buffer.putFloat(addr, data);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public long readF32(int addr) {
-        try {
-            return buffer.getInt(addr);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public float readFloat(int addr) {
-        try {
-            return buffer.getFloat(addr);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public void writeF64(int addr, double data) {
-        try {
-            buffer.putDouble(addr, data);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public double readDouble(int addr) {
-        try {
-            return buffer.getDouble(addr);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public long readF64(int addr) {
-        try {
-            return buffer.getLong(addr);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public void zero() {
-        this.fill((byte) 0);
-    }
-
-    public void fill(byte value) {
-        fill(value, 0, buffer.capacity());
-    }
+    void writeString(int offset, String data, Charset charSet);
+
+    void writeString(int offset, String data);
+
+    String readString(int addr, int len);
+
+    String readString(int addr, int len, Charset charSet);
+
+    void writeCString(int offset, String str);
+
+    void writeCString(int offset, String str, Charset charSet);
+
+    String readCString(int addr, Charset charSet);
+
+    String readCString(int addr);
+
+    void write(int addr, byte[] data);
```

Comment:
It would be better to make these default methods on the interface, as it's unlikely that other implementations of the interface will need to customize them.

Reply:
All the implementations relying on other methods in the interface are now `default`.